backslash/contrib/slash_plugin.py

```diff
@@ -145,9 +145,35 @@ def session_interrupt(self):
         if self.session is not None:
             self.session.report_interrupted()
 
+    @slash.plugins.register_if(_HAS_REPORT_UPCOMING_TESTS)
+    @handle_exceptions
+    def report_upcoming_tests(self):
```

Comment:
Changing it to `tests_collected` will also allow you to pass the tests as an argument instead of storing it globally...